app/parsers/bulkrax/bagit_parser.rb

```diff
@@ -36,39 +44,41 @@ def import_fields
       end.flatten.compact.uniq
     end
 
-    # Assume a single metadata record per path
-    # Create an Array of all metadata records, one per file
+    # Create an Array of all metadata records
     def records(_opts = {})
       raise StandardError, 'No BagIt records were found' if bags.blank?
       @records ||= bags.map do |bag|
         path = metadata_path(bag)
         raise StandardError, 'No metadata files were found' if path.blank?
         data = entry_class.read_data(path)
-        data = entry_class.data_for_entry(data, source_identifier, self)
-        data[:file] = bag.bag_files.join('|') unless importerexporter.metadata_only?
-        data
+        get_data(bag, data)
       end
+
+      @records = @records.flatten
     end
 
-    # Find or create collections referenced by works
-    # If the import data also contains records for these works, they will be updated
-    # during create works
-    def create_collections
-      collections.each_with_index do |collection, index|
-        next if collection.blank?
-        metadata = {
-          title: [collection],
-          work_identifier => [collection],
-          visibility: 'open',
-          collection_type_gid: Hyrax::CollectionType.find_or_create_default_collection_type.gid
-        }
-        new_entry = find_or_create_entry(collection_entry_class, collection, 'Bulkrax::Importer', metadata)
-        ImportCollectionJob.perform_now(new_entry.id, current_run.id)
-        increment_counters(index, collection: true)
+    def get_data(bag, data)
+      if entry_class == CsvEntry
+        data = data.map do |data_row|
+          record_data = entry_class.data_for_entry(data_row, source_identifier, self)
+          next record_data if importerexporter.metadata_only?
+
+          record_data[:file] = bag.bag_files.join('|') if Hyrax.config.curation_concerns.include? record_data[:model]&.constantize
+          record_data
+        end
+      else
+        data = entry_class.data_for_entry(data, source_identifier, self)
+        data[:file] = bag.bag_files.join('|') unless importerexporter.metadata_only?
       end
+
+      data
     end
 
     def create_works
+      entry_class == CsvEntry ? super : create_rdf_works
```

Comment:
we want to be able to create more than works with a bagit csv.

Reply:
(Disclaimer: I know very little about bagit) -- if there's this much overlap between the bagit and csv parsers, should the bagit parser inherit from the csv parser..? 🤔  Instead of pushing "csv behavior" into the application parser?

Reply:
I thought about this too. bagit itself doesn't inherit so much from the csv parser, it's just the csv format of bagit that inherits from it. the rdf portion does it's own thing...which is why stuff like this exists.

although moving the csv only stuff back into the csv parser and having bagit inherit from it would mean that overrides only need to exist in the bagit parser. so that may be worth a shot.